### PR TITLE
Reduce cognitive complexity

### PR DESCRIPTION
**Pull request by tompsota**
Signed-off-by: Tomas Psota <to.psota@gmail.com>

**What this PR does / why we need it**:
Reduces cognitive complexity of func CreateSysprepDisks and therefore fixes following sonarcloud issue: https://sonarcloud.io/project/issues?id=kubevirt_kubevirt&open=AXewXGpXmBrudGDaPwaN&resolved=false

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/config/sysprep.go
Comment (rhrazdil) [inline pkg/config/sysprep.go:80]: I may read better if this condition remained in the `CreateSysprepDisks`

**Review by rhrazdil — CHANGES_REQUESTED** [review]
Looking better, thanks! I have one comment

File: pkg/config/sysprep.go
Comment (rhrazdil) [inline pkg/config/sysprep.go:70]: Could you please also add a description to the commit message saying what the commit does

File: pkg/config/sysprep.go
Comment (maiqueb) [inline pkg/config/sysprep.go:73]: Why don't you just `return createSysprepDisk(volume)` ? afaiu, you're not doing any error handling - not even logging.

Just return it.

File: pkg/config/sysprep.go
Comment (maiqueb) [inline pkg/config/sysprep.go:84]: I'd remove this empty line.

File: pkg/config/sysprep.go
Comment (maiqueb) [reply pkg/config/sysprep.go:80]: +1.

It would help if you check the volume's "ability" to create a sysprep disk on the loop, and in the generator function, simply create it.

I recommend something like (in the loop's body):
```golang
if volume.Sysprep != nil && sysprepVolumeHasContents(*volume.Sysprep) {
...
}
```

Or better yet: export that if condition to a predicate, and name it accordingly.

File: pkg/config/sysprep.go
Comment (maiqueb) [inline pkg/config/sysprep.go:81]: Is there a reason for initializing this variable ? 

Why don't you just 

```golang
filesPath, err := getFilesLayout(sysprepSourcePath)
if err != nil {
	return err
}
```

File: pkg/config/sysprep.go
Comment (maiqueb) [inline pkg/config/sysprep.go:101]: I would export this into a function. Just be sure to name it well.

**Review by maiqueb — CHANGES_REQUESTED** [review]
While this is for sure a nice improvement of the current code base, I'm sure you can make it more readable.

Thanks.

File: pkg/config/sysprep.go
Comment (kwiesmueller) [reply pkg/config/sysprep.go:80]: Also a general good practice with checks like these is to return early.
So instead if indenting everything into this if, just invert the check:
```
if !sysprepVolumeHasContents(*volume.Sysprep) {
  return nil
}
```
But I tend to agree with the previous comments. EIther move it up or put it into a `shouldCreateSysprep(volume)` and put it here. Then you can also drop the nil check from above down into that and simplify the look even further.

File: pkg/config/sysprep.go
Comment (tompsota) [reply pkg/config/sysprep.go:73]: If I just return, the behavior of the function might change because this block of code is inside for loop. Your suggestion would make sense only if the conditions `if volume.Sysprep != nil` and `if sysprepVolumeHasContents(*volume.Sysprep)` are evaluated to `true` only once. If this is the case, I'll simplify this to just return. Otherwise it will be imo better to keep it as it is.

File: pkg/config/sysprep.go
Comment (kwiesmueller) [inline pkg/config/sysprep.go:69]: Now you can reduce complexity even further:
```suggestion
		if !shouldCreateSysprepDisk(volume) {
		  return nil
		}
		if err := createSysprepDisk(volume); err != nil {
			return err
		}
		
```

Comment (rhrazdil) [reply pkg/config/sysprep.go:69]: Hm, this may not work IIUC. We loop over all VM volumes, so for example, if there are two disks and the sysprep one happens to be the second, than with this change we would return nil and never created the sysprep disk, right?

Comment (kwiesmueller) [reply pkg/config/sysprep.go:69]: Okay, then `continue` instead

Comment (tompsota) [reply pkg/config/sysprep.go:69]: Your suggestion makes sense and with `continue` it'll definitely work, but is it gonna improve the code? The cognitive complexity would go down, but imo readability of the code will be worse. That's why I'd rather keep this condition unchanged.

Comment (kwiesmueller) [reply pkg/config/sysprep.go:69]: The complexity rule exists for readability.
Exiting early is a common practice especially in Golang and should be used where possible to reduce indentation where possible, imo.

Comment (kwiesmueller) [reply pkg/config/sysprep.go:69]: A good source on some proverbs for Go (but honestly any other language): https://dave.cheney.net/2020/02/23/the-zen-of-go

**Comment by rhrazdil** [comment]
Looking way better than the original code, thanks
/ltgm

**Comment by kwiesmueller** [comment]
Thanks for the changes!
/lgtm

File: pkg/config/sysprep.go
Comment (AlonaKaplan) [inline pkg/config/sysprep.go:83]: To avoid unnecessary copy of the whole volume, I would pass to this method just the volume name and not the whole volume.

File: pkg/config/sysprep.go
Comment (AlonaKaplan) [inline pkg/config/sysprep.go:96]: same

File: pkg/config/sysprep.go
Comment (AlonaKaplan) [inline pkg/config/sysprep.go:79]: To avoid unnecessary copy of the whole volume, I would pass to this method just `volume.Sysprep` and not the whole volume.

**Comment by qinqon** [comment]
"""
/hold
until emergency CI fixes of #5207 are merged
"""

**Comment by phoracek** [comment]
/hold cancel

The critical PR was merged

**Pull request merged**
